### PR TITLE
Add static & dockerized targets to Makefile.

### LINKED ISSUE #32: Static builds
Hey there :wave:, do we need static binary builds of conmon? We currently have some dependencies in:

```
> ldd bin/conmon
        linux-vdso.so.1 (0x00007fffafde9000)
        libglib-2.0.so.0 => /usr/lib64/libglib-2.0.so.0 (0x00007f6325932000)
        libsystemd.so.0 => /usr/lib64/libsystemd.so.0 (0x00007f6325876000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f63256b1000)
        libpcre.so.1 => /usr/lib64/libpcre.so.1 (0x00007f6325621000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f6325600000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f63255f5000)
        liblzma.so.5 => /usr/lib64/liblzma.so.5 (0x00007f63255bc000)
        liblz4.so.1 => /usr/lib64/liblz4.so.1 (0x00007f632559c000)
        libcap.so.2 => /usr/lib64/libcap.so.2 (0x00007f6325595000)
        libselinux.so.1 => /lib64/libselinux.so.1 (0x00007f6325569000)
        libgcrypt.so.20 => /usr/lib64/libgcrypt.so.20 (0x00007f6325449000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6325a76000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f6325444000)
        libgpg-error.so.0 => /usr/lib64/libgpg-error.so.0 (0x00007f632541e000)
```

Feel free to close if this is out of scope.

### PR DESCRIPTION
**Pull request by mgoltzsche**
Closes #32.

I've added a `dockerized` target for convenience as well.

Signed-off-by: Max Goltzsche <max.goltzsche@gmail.com>

### CODE REVIEW

**Comment by haircommander** [comment]
nice @mgoltzsche thanks for the fix. I think @rhatdan will ask you to change it to `containerized` and switch the manager to podman...

**Comment by haircommander** [comment]
bot, add author to whitelist

**Comment by mgoltzsche** [comment]
I just renamed the target to `containerized` and added flags (`-s -w`) to shrink the binary from 3.6M to 870K.

**Comment by haircommander** [comment]
s/docker/podman then LGTM :smile:

**Comment by mgoltzsche** [comment]
oups, I've added it conditionally now.

**Comment by haircommander** [comment]
sorry for the stream of nits, but you missed the build command

**Comment by mgoltzsche** [comment]
I just noticed, sorry. Fixed.

**Comment by haircommander** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM, thanks @mgoltzsche!

**Comment by mgoltzsche** [comment]
Wait, now that it could be built using podman optionally I should make the user flag optional as well :/ ...

**Comment by mgoltzsche** [comment]
Okay, now it should be mergeable finally.

File: Makefile [hunk 1]
```diff
@@ -43,6 +43,30 @@ else ifeq ($(shell $(PKG_CONFIG) --exists libsystemd && echo "0" || echo "1"), 0
 	override CFLAGS += $(shell $(PKG_CONFIG) --cflags libsystemd) -D USE_JOURNALD=0
 endif
 
+define DOCKERFILE
+	FROM alpine:3.9
```
Comment (TomSweeneyRedHat) [inline Makefile]: nit of a nit and sorry that I didn't spot earlier.  Could we do `:latest` here for the tag?

Comment (mgoltzsche) [reply Makefile]: done

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by mgoltzsche** [comment]
Damn, tons of scatterbrained hostility errors... yesterday wasn't my day...

The Dockerfile resolution error obviously happened because of invalid parameters (I wonder how this worked on my machine yesterday) - I ll fix it.
About the SE-Linux issue: Simply building the binary within the Dockerfile build and then copy it to the host should solve the issue without making the container privileged, shouldn't it?

**Comment by mgoltzsche** [comment]
This should work now properly with docker or podman and SE linux enabled as well.

@vrothberg would you mind testing it again on your machine?

Also I'd be glad if this would be added to your CI pipeline somehow.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**